STACpopulator/input.py

```diff
@@ -91,89 +59,92 @@ def __init__(
         :param thredds_catalog_url: the URL to the THREDDS catalog to ingest
         :type thredds_catalog_url: str
         :param depth: Maximum recursive depth for the class's generator. Setting 0 will return only datasets within the
-          top-level catalog. If None, depth is set to 1000, defaults to None
+          top-level catalog. If None, depth is unlimited.
         :type depth: int, optional
         """
         super().__init__()
-        self._max_depth = depth if depth is not None else 1000
-        self._depth = 0
-
-        self.thredds_catalog_URL = self.validate_catalog_url(thredds_catalog_url)
-
-        self.catalog = TDSCatalog(self.thredds_catalog_URL)
-        self.catalog_head = self.catalog
-        self.links.append(self.magpie_collection_link())
-
-    def validate_catalog_url(self, url: str) -> str:
-        """Validate the user-provided catalog URL.
-
-        :param url: URL to the THREDDS catalog
-        :type url: str
-        :raises RuntimeError: if URL is invalid or contains query parameters.
-        :return: a valid URL
-        :rtype: str
-        """
-        if url_validate(url):
-            if "?" in url:
-                raise RuntimeError("THREDDS catalog URL should not contain query parameter")
-        else:
-            raise RuntimeError("Invalid URL")
-
-        return url.replace(".html", ".xml") if url.endswith(".html") else url
-
-    def magpie_collection_link(self) -> pystac.Link:
-        """Create a PySTAC Link for the collection that is used by Cowbird and Magpie.
-
-        :return: A PySTAC Link
-        :rtype: pystac.Link
-        """
-        url = self.thredds_catalog_URL
-        parts = url.split("/")
-        i = parts.index("catalog")
-        service = parts[i - 1]
-        path = "/".join(parts[i + 1 : -1])
-        title = f"{service}:{path}"
-        return pystac.Link(rel="source", target=url, media_type="text/xml", title=title)
-
-    def reset(self) -> None:
-        """Reset the generator."""
-        self.catalog_head = self.catalog
+        self.depth = depth
+        self.session = session or requests.Session()
+        session_manager.set_session_options(**vars(self.session))
+        try:
+            self.catalog = TDSCatalog(thredds_catalog_url)
+        except requests.exceptions.RequestException as exc:
+            LOGGER.error(
+                "Could not access THREDDS host. Not reachable [%s] due to [%s]", thredds_catalog_url, exc, exc_info=exc
+            )
+        self.links.append(pystac.Link(rel="source", target=self.catalog.catalog_url, media_type="application/xml"))
```

Comment:
Note that the `title` is removed due to: https://github.com/crim-ca/stac-populator/pull/88/files#r2114678977